### PR TITLE
PD: Implement 1D intersection for attachments

### LINKED ISSUE #6870: [Feature Request] New Datum Line based on Intersection between two Planes
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?t=62149

### Subproject(s) affected?

PartDesign

### Idea description

Hi, 
I'm requesting an old feature discussed on FreeCAD forum (https://forum.freecadweb.org/viewtopic.php?t=62149)
Would it be possible to create a Datum Line from the intersection of two datum planes? 
There is a macro to do this ([datumplane_intersection.FCMacro](https://forum.freecadweb.org/download/file.php?id=164282)) but it would be best for it to be imbedded into Part Design Workbench, thus making it parametric.

Thanks for your hard work!

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by LemonBoy**
Extract a line from the intersection of two faces.

Closes #6870

Please give it a spin, I didn't manage to break it after some use...

### CODE REVIEW

**Comment by luzpaz** [comment]
Note: macOS build is failing

**Comment by LemonBoy** [comment]
> Note: macOS build is failing

I really don't understand what the error is, the relevant message seems to be 

> FAILED: src/Mod/Sketcher/App/CMakeFiles/Sketcher.dir/SketchPyImp.cpp.o 

but I cannot spot any compiler/linker error regarding that object file.

**Comment by FlachyJoe** [comment]
MacOS CI certainly has a problem, I have a PR for which it fails on my repo but works on FreeCAD's one. :roll_eyes:

**Review by FlachyJoe — COMMENTED** [review]
Tested: it works as expected.
You eventually can join the `if` blocks which raise the same error.

File: src/Mod/Part/App/Attacher.cpp [hunk 1]
```diff
@@ -1886,6 +1887,33 @@ Base::Placement AttachEngineLine::calculateAttachedPlacement(const Base::Placeme
...
+            if (shapes[1]->IsNull())
+                throw Base::ValueError("Null shape in AttachEngineLine::calculateAttachedPlacement()!");
+
+            const TopoDS_Face &face1 = TopoDS::Face(*(shapes[0]));
+            const TopoDS_Face &face2 = TopoDS::Face(*(shapes[1]));
+
+            Handle(Geom_Surface) hSurf1 = BRep_Tool::Surface(face1);
+            Handle(Geom_Surface) hSurf2 = BRep_Tool::Surface(face2);
+            GeomAPI_IntSS intersector(hSurf1, hSurf2, Precision::Confusion());
+            if (!intersector.IsDone())
```
[the remark this answers is not in the document]
Comment (LemonBoy) [reply src/Mod/Part/App/Attacher.cpp:1904]: Not sure if we want a different error here, after all the intersection didn't fail but yielded zero or more than one curves.
The `mm1Proximity` case suggests another strategy: raise an exception if the number of lines is zero and consider only the first curve when lines is greater or equal to one, of course it does warn the user about the potential ambiguity.

Another question, should all the exception messages be prefixed with "AttachEngineLine::calculateAttachedPlacement" ?

[the remark this answers is not in the document]
Comment (adrianinsaval) [reply src/Mod/Part/App/Attacher.cpp:1904]: It can't hurt to give a different message IMO

**Pull request merged**